waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -45,11 +52,38 @@ proc init*(T: type WakuMessageStore, db: SqliteDatabase): MessageStoreResult[T]
   if prepare.isErr:
     return err("failed to prepare")
 
-  let res = prepare.value.exec(())
-  if res.isErr:
+  let prepareRes = prepare.value.exec(())
+  if prepareRes.isErr:
     return err("failed to exec")
 
-  ok(WakuMessageStore(database: db))
+  # count messages in DB
+  var numMessages: int64
+  proc handler(s: ptr sqlite3_stmt) = 
+    numMessages = sqlite3_column_int64(s, 0)
+  let countQuery = "SELECT COUNT(*) FROM " & TABLE_TITLE
+  let countRes = db.query(countQuery, handler)
+  if countRes.isErr:
+    return err("failed to count number of messages in DB")
+
+  debug "number of messages in sqlite database", messageNum=numMessages
+
+  ok(WakuMessageStore(database: db,
+                      numMessages: int(numMessages),
+                      maxNumMessages: 2000,
+                      truncateWindowSize: int(maxNumMessages * truncateWindowPercentage/100)))
+
+proc truncate(db: WakuMessageStore): MessageStoreResult[void] =
```

Comment:
A bit of a nitpick, but doesn't `truncate` in SQL terms usually mean deleting _all_ rows from a table (not just the oldest)? In other words, if we want to add such a function in future we may run into clashing terminology. Perhaps calling this something closer to `deleteOldest` or `resize` could avoid confusion in future.

Reply:
Ah, yes! That's on me. :)